### DISCUSSION TITLE
Support security scanning for openEuler Linux

### DISCUSSION THREAD

**Discussion by wjunLu** [aquasecurity/trivy#6400] [Ideas]
## What's openEuler?
openEuler is an open source, free Linux distribution platform. The platform provides an open community for global developers to build an open, diversified, and architecture-inclusive software ecosystem. openEuler is also an innovative platform that
encourages everyone to propose new ideas, explore new approaches, and practice new solutions.

Learn more, please visit https://www.openeuler.org/en/

## Trivy does not support openEuler
We can see that the operating systems currently supported by trivy for security detection does not include openEuler(see https://aquasecurity.github.io/trivy/v0.50/docs/coverage/os/).  

![image](https://github.com/aquasecurity/trivy/assets/135617475/1ff278ab-c29d-4578-82d2-ecad2613f90c)

## Support openEuler
Now, openEuler has 2,345,659 users, 18,072 contributors and 1,501 organization members(see https://datastat.openeuler.org/en/overview). It is necessary to support such a very mature open source operating system.

### Target

Container Image

### Scanner

Vulnerability

**Comment by knqyf263** [comment]
We would like to see the level of demand from the community.

>Now, openEuler has 2,345,659 users, 18,072 contributors and 1,501 organization members(see https://datastat.openeuler.org/en/overview). It is necessary to support such a very mature open source operating system.

If they're not interested in scanning openEuler with Trivy, we may not want to add support for it.

**Reply by wjunLu** [reply]
As a committer of the openEuler community, I can say that the demand for using trivy to scan openEuler images is very high!

1. openEuler has some repositories of container image in [hub.oepkgs.net](https://hub.oepkgs.net/harbor/projects/10/repositories/openeuler/artifacts-tab?globalSearch=openeuler), which use the harbor-trivy-adapter plugin to perform vulnerability scanning of openEuler container images. However, trivy does not support to scan openEuler, making scanning results meaningless.

2. We are considering to build a new hub, and planing to use trivy as the security scanner for openEuler.

We very much hope that trivy can support for scanning openEuler!

**Reply by knqyf263** [reply]
>We are considering to build a new hub, and planing to use trivy as the security scanner for openEuler.

Cool. Then, we'll look into it. Do you mind adding openEuler [as an adopter](https://github.com/aquasecurity/trivy/discussions/categories/adopters) after the new hub is out?

**Reply by wjunLu** [reply]
Yes, of course!

**Reply by fordnbc** [reply]
please add support for openeuler and openeuler     embeeded as well. Today as Centos not support any longer, more lagacy servers will migrate to open euler serias

**Reply by wjunLu** [reply]
We have submitted PRs, please check:
- https://github.com/aquasecurity/trivy/pull/6475
- https://github.com/aquasecurity/trivy-db/pull/397
- https://github.com/aquasecurity/vuln-list-update/pull/284

**Comment by itaysk** [comment]
thanks for your suggestion and contributions. I can confirm that supporting OpenEuler is within scope, but we are currently busy adding support for other ecosystems and features as prioritized by for our [Partners](https://trivy.dev/latest/partners/). We will keep monitoring community interest and resume this task when we have more capacity.

**Reply by wjunLu** [reply]
@itaysk Thank you!
Please feel free to leave your reply here when you have time to handle openEuler.

**Reply by wjunLu** [reply]
Hi!
Just a friendly follow-up to check whether there’s any update or available bandwidth to review this PR.

Thanks again for your time and for considering the openEuler support!
